paddle/fluid/framework/ir/conv_elementwise_add_mkldnn_fuse_pass.cc

```diff
@@ -0,0 +1,246 @@
+      elementwise_add_op->LinksTo({out_var});
+
+      return out_var;
+    };
+  }
+};
+
+Node* GetNodeFromSubgraph(const GraphPatternDetector::subgraph_t& subgraph,
+                          std::shared_ptr<patterns::Pattern> pattern,
+                          const std::string& op_name) {
+  PADDLE_ENFORCE(subgraph.count(pattern->retrieve_node(op_name)),
+                 "Node not found for PDNode %s", pattern->node_name(op_name));
+  Node* var = subgraph.at(pattern->retrieve_node(op_name));
+  PADDLE_ENFORCE(var, "node %s not exists in the sub-graph");
+
+  return var;
+}
+
+void LinkNodes(Node* from, Node* to) {
+  from->outputs.push_back(to);
+  to->inputs.push_back(from);
+}
+
+template <typename IT, typename FindFunc, typename ReplaceFunc>
+void ReplaceAllOccurances(IT s, IT e, FindFunc f, ReplaceFunc r) {
+  if (s == e) return;
+
+  auto it = std::find_if(s, e, f);
+
+  if (it != e) {
+    r(*it);
+  }
+
+  it++;
+  ReplaceAllOccurances(it, e, f, r);
+}
+
+void CorrectGraphEdges(Graph* graph, Node* from, Node* to) {
+  for (auto& node : GraphTraits::DFS(*graph)) {
+    auto same = std::find_if(std::begin(node.inputs), std::end(node.inputs),
+                             [from](Node* n) { return n == from; });
+
+    if (same != std::end(node.inputs)) {
+      LinkNodes(to, &node);
+
+      auto inputs = node.Op()->Inputs();
+
+      using input_type = VariableNameMap::value_type;
+
+      ReplaceAllOccurances(
+          std::begin(inputs), std::end(inputs),
+          [from](const input_type& i) -> bool {
+            auto params = i.second;
+            auto pi =
+                std::find_if(std::begin(params), std::end(params),
+                             std::bind(std::equal_to<std::string>(),
+                                       from->Name(), std::placeholders::_1));
+            return pi != std::end(params);
+          },
+          [to, &node](const input_type& i) {
+            node.Op()->SetInput(i.first, {to->Name()});
+          });
+    }
+  }
+}
+}  // namespace patterns
+using graph_ptr = std::unique_ptr<ir::Graph>;
+
+graph_ptr ConvElementwiseAddMKLDNNFusePass::ApplyImpl(graph_ptr graph) const {
+  FusePassBase::Init("conv_elementwise_add_mkldnn_fuse_pass", graph.get());
+
+  GraphPatternDetector gpd;
+  auto pattern = gpd.mutable_pattern();
+  auto pattern_ptr = std::make_shared<patterns::Pattern>(pattern, name_scope_);
+
+  patterns::Conv conv_pattern;
+  auto conv_output = conv_pattern(pattern_ptr)();
+
+  patterns::ElementwiseAdd elementwise_add_pattern;
+  elementwise_add_pattern(pattern_ptr)(conv_output);
+
+  conv_output->AsIntermediate();
+
+  auto fuse_conv = [&conv_pattern](Graph* g, Node* conv_input, Node* conv_bias,
+                                   Node* conv_filter, Node* conv_output,
+                                   Node* elementwise_add_x) {
+    OpDesc op_desc;
+    op_desc.SetType(conv_pattern.op_name());
+
+    op_desc.SetInput(conv_pattern.input_name(), {conv_input->Name()});
```

Comment:
Could you use `SetInput` macro to simplify these codes?

Reply:
I couldn't find this macro.